lib/rspec/matchers/built_in/raise_error.rb

```diff
@@ -166,11 +177,37 @@ def warn_about_bare_error
                          "_positives = :nothing`")
         end
 
+        def warn_about_negative_nil_error
+          handle_warning("Using `raise_error(nil)` risks false positives, since literally " \
+                         "any other error would cause the expectation to pass, including " \
+                         "those raised by Ruby (e.g. `NoMethodError`, `NameError` and " \
+                         "`ArgumentError`), meaning the code you are intending to test may " \
+                         "not even get reached. " \
+                         "Instead consider using `expect { }.not_to raise_error` or " \
+                         "`expect { }.to raise_error(DifferentSpecificErrorClass)`. " \
+                         "This message can be suppressed by setting: " \
+                         "`RSpec::Expectations.configuration.on_potential_false" \
+                         "_positives = :nothing`")
+        end
```

Comment:
Would this not be the same as the original negative error warning?

Reply:
I mean could you not do this with `warn_about_negative_false_positive!("raise_error(nil)")`

Reply:
I've added this distinctive message intentionally. First off, I don't think anyone would deliberately write `.not_to raise_error(nil)`, so there should be some kind of mistake, and a regular message telling that didn't pass anything to `raise_error` and they'd better use `raise_error(DifferentSpecificErrorClass)` might be confusing and would suggest an idea of an RSpec bug, since they obviously passed something. Another point was to make it more visible to remind us that we planned to switch it to raise an error eventually.
Do you think it's reasonable enough to keep a separate message?

Reply:
> I've added this distinctive message intentionally.

Except it's not distinct? It's identical except for swapping `#{expression}` for `raise_error(nil)` so why not set expression to `"raise_error(nil)"` and generate the same string using the existing method?

Reply:
Sorry for the delay in getting your idea. Fixed, it turns out to be more concise the way you suggested 👍